migrations/lib/importer/topological_sorter.rb

```diff
@@ -28,19 +31,48 @@ def sort
         end
       end
 
-      raise "Circular dependency detected" if result.size < @classes.size
+      raise "Circular dependency detected" if result.size < @filtered_classes.size
 
       result
     end
 
     private
 
+    def filter_classes
+      classes_to_include =
+        (
+          if @only.empty?
+            @classes
+          else
+            @classes.select { |klass| @only.include?(klass.name.demodulize.underscore) }
+          end
+        )
+      classes_to_include =
+        classes_to_include.reject { |klass| @skip.include?(klass.name.demodulize.underscore) }
+
+      with_dependencies = Set.new(classes_to_include)
+      classes_to_include.each { |klass| add_dependencies(klass, with_dependencies) }
+
+      with_dependencies.to_a
+    end
+
+    def add_dependencies(klass, included_set)
+      dependencies = klass.dependencies || []
+      dependencies.each do |dependency|
+        next if @skip.include?(dependency.name.demodulize.underscore)
+        next if included_set.include?(dependency)
+
+        included_set << dependency
+        add_dependencies(dependency, included_set)
+      end
+    end
```

Comment:
Tiny nit: Could we build a one-time map of the demodulized underscored step class names (klass.name.demodulize.underscore) instead of recomputing them each time?